azurelinuxagent/common/version.py

```diff
@@ -183,15 +184,22 @@ def get_lis_version():
         return COMMAND_FAILED
 
 def has_logrotate():
+    basic_version_regex = r"\d(?:\.\d+)*"
+    regex = r"logrotate (?P<version>{ver})".format(ver=basic_version_regex)
+
     try:
-        logrotate_version = shellutil.run_command(["logrotate", "--version"]).split("\n")[0]
-        return logrotate_version
-    except shellutil.CommandError:
-        # A non-zero return code means that logrotate isn't present on 
-        # the system; --version shouldn't fail otherwise.
-        return COMMAND_ABSENT
-    except Exception:
-        return COMMAND_FAILED
+        cmd_output = shellutil.run_command(["logrotate", "--version"], stderr=subprocess.STDOUT)
+        match = re.search(regex, cmd_output)
+        if match:
+            return match.group("version")
+
+    except Exception as e:
+        if is_file_not_found_error(e):
+            # run_command actually throws a FileNotFound if there
+            # is no executable corresponding to the command given.
+            return COMMAND_ABSENT
```

Comment:
"Absent" and "Failed" don't seem to be very useful. Why do we even attempt to parse instead of just logging what the command returns?

Reply:
In the case that logrotate is not present on the system, there is no output, and a file not found error is thrown. I suppose the regex matching doesn't belong in the try clause... But with no output, we have to return something. 

Additionally, certain logrotate versions have very verbose outputs, e.g.:

```bash
logrotate 3.14.0

    Default mail command:       /usr/bin/mail
    Default compress command:   /bin/gzip
    Default uncompress command: /bin/gunzip
    Default compress extension: .gz
    Default state file path:    /var/lib/logrotate/status
    ACL support:                yes
    SELinux support:            yes
```

We don't want to log all of this in the OSInfo message, right?

Reply:
If the output changes per distro then attempting to parse may also have issues. Let's remove this info from the log and come back to it on V3